.gitignore

```diff
@@ -7,3 +7,4 @@ tmp
 *.lst
 *.o
 *.obj
+build
```

Comment:
Shown as "No newline at end of file".
At least, add the newline.

Reply:
(Strange, I cannot anwer per file comment here anymore.)

_io.d_
I agree now. The offset semantic should be explicit enough with the code and comments in the Input constructor:
```
    this(string name, const(char)[] text, bool useOffset = false) @nogc nothrow
    {
        ...
        this.useOffset = useOffset;
        if (useOffset) 
        {
            this.line = 0;
            this.col = 0;
        }
        else 
        {
            this.line = 1;
            this.col = 1;
        }
    }
```
_main.d_
I think the other switches should work. Only the --verbose is problematic. I'm not sure, if I always can skip the output when parsing.
Could we ensure the verbose output is always going to stdout?
Because the errors with offset position are always going to stderr, so I could ignore the verbose output very easily.

_.gitignore_
Of course.